homeassistant/components/frontier_silicon/media_player.py

```diff
@@ -64,11 +71,12 @@ async def async_setup_platform(
     name = config.get(CONF_NAME)
 
     try:
-        async_add_entities(
-            [AFSAPIDevice(f"http://{host}:{port}/device", password, name)], True
+        webfsapi_url = await AFSAPI.get_webfsapi_endpoint(
+            f"http://{host}:{port}/device"
         )
-        _LOGGER.debug("FSAPI device %s:%s -> %s", host, port, password)
-    except requests.exceptions.RequestException:
+        afsapi = AFSAPI(webfsapi_url, password)
+        async_add_entities([AFSAPIDevice(name, afsapi)], True)
```

Comment:
Please only wrap the line that can raise in the try... except block. We can move the other lines below of we return after logging the error.